File: quan-java/config/src/test/java/quan/config/item/EquipConfig.java
package quan.config.item;

import java.util.*;
import com.alibaba.fastjson.*;
import quan.config.*;

/**
* 装备1,装备2<br/>
* Created by 自动生成
*/
public class EquipConfig extends ItemConfig {

    //部位
    protected final int position;

    //颜色
    protected final int color;


    public EquipConfig(JSONObject json) {
        super(json);

        this.position = json.getIntValue("position");
        this.color = json.getIntValue("color");
    }

    /**
     * 部位
     */
    public final int getPosition() {
        return position;
    }

    /**
     * 颜色
     */
    public final int getColor() {
        return color;
    }


    @Override
    protected EquipConfig create(JSONObject json) {
        return new EquipConfig(json);
    }

    @Override
    public String toString() {
        return "EquipConfig{" +
                "id=" + id +
                ",name='" + name + '\'' +
                ",type=" + type +
                ",reward=" + reward +
                ",list=" + list +
                ",set=" + set +
                ",map=" + map +
                ",effectiveTime='" + effectiveTime$Str + '\'' +
                ",position=" + position +
                ",color=" + color +
                '}';

    }


    public static class self {

        private self() {
        }

        private static volatile List<EquipConfig> configs = new ArrayList<>();

        //ID
        private static volatile Map<Integer, EquipConfig> idConfigs = new HashMap<>();

        //部位
        private static volatile Map<Integer, List<EquipConfig>> positionConfigs = new HashMap<>();

        public static List<EquipConfig> getConfigs() {
            return configs;
        }

        public static Map<Integer, EquipConfig> getIdConfigs() {
            return idConfigs;
        }

        public static EquipConfig getById(int id) {
            return idConfigs.get(id);
        }

        public static Map<Integer, List<EquipConfig>> getPositionConfigs() {
            return positionConfigs;
        }

        public static List<EquipConfig> getByPosition(int position) {
            return positionConfigs.getOrDefault(position, Collections.emptyList());
        }


        @SuppressWarnings({"unchecked"})
        public static List<String> index(List<EquipConfig> configs) {
            Map<Integer, EquipConfig> idConfigs = new HashMap<>();
            Map<Integer, List<EquipConfig>> positionConfigs = new HashMap<>();

            List<String> errors = new ArrayList<>();
            EquipConfig oldConfig;

            for (EquipConfig config : configs) {
                oldConfig = idConfigs.put(config.id, config);
                if (oldConfig != null) {
                    String repeatedConfigs = config.getClass().getSimpleName();
                    if (oldConfig.getClass() != config.getClass()) {
                        repeatedConfigs += "," + oldConfig.getClass().getSimpleName();
                    }
                    errors.add(String.format("配置[%s]有重复数据[%s = %s]", repeatedConfigs, "id", config.id));
                }

                positionConfigs.computeIfAbsent(config.position, k -> new ArrayList<>()).add(config);
            }

            configs = Collections.unmodifiableList(configs);
            idConfigs = unmodifiableMap(idConfigs);
            positionConfigs = unmodifiableMap(positionConfigs);

            EquipConfig.self.configs = configs;
            EquipConfig.self.idConfigs = idConfigs;
            EquipConfig.self.positionConfigs = positionConfigs;

            return errors;
        }

    }

}

File: quan-cs/Test/Config/Item/EquipConfig.cs
using System.Collections.Generic;
using System.Collections.Immutable;
using Newtonsoft.Json.Linq;
using Quan.Common;
using Quan.Config;

namespace Test.Config.Item
{
    /// <summary>
	/// 装备1,装备2<br/>
	/// 自动生成
	/// </summary>
    public class EquipConfig : ItemConfig
    {
        /// <summary>
        /// 部位
        /// </summary>
        public readonly int Position;

        /// <summary>
        /// 颜色
        /// </summary>
        public readonly int Color;


        public EquipConfig(JObject json) : base(json)
        {
            Position = json["position"]?.Value<int>() ?? default;
            Color = json["color"]?.Value<int>() ?? default;
        }

        protected override ConfigBase Create(JObject json)
        {
            return new EquipConfig(json);
        }


        public override string ToString()
        {
            return "EquipConfig{" +
                   "Id=" + Id.ToString2() +
                   ",Name='" + Name + '\'' +
                   ",Type=" + Type.ToString2() +
                   ",Reward=" + Reward.ToString2() +
                   ",List=" + List.ToString2() +
                   ",Set=" + Set.ToString2() +
                   ",Map=" + Map.ToString2() +
                   ",EffectiveTime=" + EffectiveTime.ToString2() +
                   ",Position=" + Position.ToString2() +
                   ",Color=" + Color.ToString2() +
                   '}';
        }


        // 所有EquipConfig
        private static volatile IList<EquipConfig> _configs = new List<EquipConfig>();

        // ID
        private static volatile IDictionary<int, EquipConfig> _idConfigs = new Dictionary<int, EquipConfig>();

        // 部位
        private static volatile IDictionary<int, IList<EquipConfig>> _positionConfigs = new Dictionary<int, IList<EquipConfig>>();

        public new static IList<EquipConfig> GetConfigs()
        {
            return _configs;
        }

        public new static IDictionary<int, EquipConfig> GetIdConfigs()
        {
            return _idConfigs;
        }

        public new static EquipConfig GetById(int id)
        {
            _idConfigs.TryGetValue(id, out var result);
            return result;
        }

        public static IDictionary<int, IList<EquipConfig>> GetPositionConfigs()
        {
            return _positionConfigs;
        }

        public static IList<EquipConfig> GetByPosition(int position)
        {
            _positionConfigs.TryGetValue(position, out var result);
            return result ?? ImmutableList<EquipConfig>.Empty;
        }


        public static void Index(IList<EquipConfig> configs)
        {
            IDictionary<int, EquipConfig> idConfigs = new Dictionary<int, EquipConfig>();
            IDictionary<int, IList<EquipConfig>> positionConfigs = new Dictionary<int, IList<EquipConfig>>();

            foreach (var config in configs)
            {
                ConfigBase.Index(idConfigs, config, config.Id);
                ConfigBase.Index(positionConfigs, config, config.Position);
            }

            configs = configs.ToImmutableList();
            idConfigs = ToImmutableDictionary(idConfigs);
            positionConfigs = ToImmutableDictionary(positionConfigs);

            _configs = configs;
            _idConfigs = idConfigs;
            _positionConfigs = positionConfigs;
        }
    }
}